frontend/src/metabase-lib/Question.ts

```diff
@@ -1286,6 +1298,211 @@ class QuestionInner {
+          case "count":
+            return [t`Count`];
+          case "cum-count":
+            return [t`Cumulative count`];
+          case "avg":
+            return [
+              t`Average of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "median":
+            return [
+              t`Median of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "distinct":
+            return [
+              t`Distinct values of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "stddev":
+            return [
+              t`Standard deviation of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "sum":
+            return [
+              t`Sum of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "cum-sum":
+            return [
+              t`Cumulative sum of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "max":
+            return [
+              t`Maximum of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          case "min":
+            return [
+              t`Minimum of `,
+              this._getFieldName(tableMetadata, aggregation[1], options),
+            ];
+          default:
+            return [formatExpression(aggregation, { tableMetadata })];
+        }
+      }),
+      "and",
+    );
+  }
+
+  private _getBreakoutDescription(tableMetadata, { breakout }, options) {
+    if (breakout && breakout.length > 0) {
+      return [
+        t`Grouped by `,
+        DESCRIPTION.joinList(
+          breakout.map(b => this._getFieldName(tableMetadata, b, options)),
+          " and ",
+        ),
+      ];
+    }
+  }
+
+  private _getFilterDescription(tableMetadata, query, options) {
+    // getFilters returns list of filters without the implied "and"
+    const filters = ["and"].concat(QUERY.getFilters(query));
+    if (filters && filters.length > 1) {
+      return [
+        t`Filtered by `,
+        this._getFilterClauseDescription(tableMetadata, filters, options),
+      ];
+    }
+  }
+
+  private _getFilterClauseDescription(tableMetadata, filter, options) {
+    if (filter[0] === "and" || filter[0] === "or") {
+      const clauses = filter
+        .slice(1)
+        .map(f => this._getFilterClauseDescription(tableMetadata, f, options));
+      return DESCRIPTION.conjunctList(clauses, filter[0].toLowerCase());
+    } else if (filter[0] === "segment") {
+      const segment = _.findWhere(tableMetadata.segments, { id: filter[1] });
+      return segment ? segment.name : "[Unknown Segment]";
+    } else if (filter[0] === "between" && filter[1][0] === "+") {
+      return this._getFieldName(tableMetadata, filter[1][1], options);
```

Comment:
Would it make sense to give names to things like `filter[1][1]`? Could make it a bit easier to follow, otherwise, people would need to recall how MBQL looks like every time